Add unique prefixes to all .targets properties and items

Sergio0694 opened the pull request:
This PR adds unique prefixes to all properties and items in the .targets file bundled with the MVVM Toolkit.
This ensures there can be no conflicts with other targets defining the same elements from other NuGet packages.

Essentially, this PR fixes this: https://github.com/Sergio0694/ComputeSharp/actions/runs/2523630796.

## PR Checklist

<!-- Please check if your PR fulfills the following requirements, and remove the ones that are not applicable to the current PR -->

- [X] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [X] Based off latest main branch of toolkit
- [X] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [X] Tested code with current [supported SDKs](../#supported)
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run _build/UpdateHeaders.bat_)
- [X] Contains **NO** breaking changes
- [X] Every new API (including internal ones) has full XML docs
- [X] Code follows all style conventions

Nirmal4G commented on CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets:
```diff
@@ -18,27 +18,27 @@
          someone is getting creative with msbuild tasks/targets this is the "most correct" assembly to check. -->
     <GetAssemblyIdentity AssemblyFiles="$([System.IO.Path]::Combine(`$([System.IO.Path]::GetDirectoryName($(CSharpCoreTargetsPath)))`,`Microsoft.Build.Tasks.CodeAnalysis.dll`))">
-      <Output TaskParameter="Assemblies" ItemName="CurrentCompilerAssemblyIdentity"/>
+      <Output TaskParameter="Assemblies" ItemName="MVVMToolkitCurrentCompilerAssemblyIdentity"/>
```
```suggestion
      <Output TaskParameter="Assemblies" ItemName="_MVVMToolkit_CurrentCompilerAssemblyIdentity"/>
```

Nirmal4G commented on CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets:
```diff
@@ -18,27 +18,27 @@
 
       <!-- Transform the resulting item from GetAssemblyIdentity into a property representing its assembly version -->
-      <CurrentCompilerVersion>@(CurrentCompilerAssemblyIdentity->'%(Version)')</CurrentCompilerVersion>
+      <MVVMToolkitCurrentCompilerVersion>@(MVVMToolkitCurrentCompilerAssemblyIdentity->'%(Version)')</MVVMToolkitCurrentCompilerVersion>
```
```suggestion
      <_MVVMToolkit_CurrentCompilerVersion>@(_MVVMToolkit_CurrentCompilerAssemblyIdentity->'%(Version)')</_MVVMToolkit_CurrentCompilerVersion>
```

Nirmal4G commented on CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets:
```diff
@@ -18,27 +18,27 @@
 
       <!-- The CurrentCompilerVersionIsNotNewEnough property can now be defined based on the Roslyn assembly version -->
-      <CurrentCompilerVersionIsNotNewEnough Condition="$([MSBuild]::VersionLessThan($(CurrentCompilerVersion), 4.0))">true</CurrentCompilerVersionIsNotNewEnough>
+      <MVVMToolkitCurrentCompilerVersionIsNotNewEnough Condition="$([MSBuild]::VersionLessThan($(MVVMToolkitCurrentCompilerVersion), 4.0))">true</MVVMToolkitCurrentCompilerVersionIsNotNewEnough>
```
```suggestion
      <_MVVMToolkit_CurrentCompilerVersionIsNotNewEnough Condition="$([MSBuild]::VersionLessThan($(_MVVMToolkit_CurrentCompilerVersion), 4.0))">true</_MVVMToolkit_CurrentCompilerVersionIsNotNewEnough>
```

Nirmal4G commented on CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets:
```diff
@@ -18,27 +18,27 @@
 
     <!-- If the Roslyn version is < 4.0, disable the source generators -->
-    <ItemGroup Condition ="'$(CurrentCompilerVersionIsNotNewEnough)' == 'true'">
+    <ItemGroup Condition ="'$(MVVMToolkitCurrentCompilerVersionIsNotNewEnough)' == 'true'">
```
```suggestion
    <ItemGroup Condition ="'$(_MVVMToolkit_CurrentCompilerVersionIsNotNewEnough)' == 'true'">
```

Nirmal4G commented on CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets:
```diff
@@ -18,27 +18,27 @@
          emitting this manually lets us customize the message to inform developers as to why exactly the generators have been
          disabled, and that the rest of the MVVM Toolkit will still keep working as intended, just without additional features. -->
-    <Warning Condition ="'$(CurrentCompilerVersionIsNotNewEnough)' == 'true'" Text="The MVVM Toolkit source generators have been disabled on the current configuration, as they need Roslyn 4.x in order to work. The MVVM Toolkit will work just fine, but features relying on the source generators will not be available."/>
+    <Warning Condition ="'$(MVVMToolkitCurrentCompilerVersionIsNotNewEnough)' == 'true'" Text="The MVVM Toolkit source generators have been disabled on the current configuration, as they need Roslyn 4.x in order to work. The MVVM Toolkit will work just fine, but features relying on the source generators will not be available."/>
```
```suggestion
    <Warning Condition ="'$(_MVVMToolkit_CurrentCompilerVersionIsNotNewEnough)' == 'true'" Text="The MVVM Toolkit source generators have been disabled on the current configuration, as they need Roslyn 4.x in order to work. The MVVM Toolkit will work just fine, but features relying on the source generators will not be available."/>
```

Nirmal4G reviewed (changes requested):
Seems like these properties are not meant to be public. So, you can use a `_` prefix to denote that. But even `_CurrentCompilerVersion` sometimes could be used elsewhere as the name is too generic. Hence, my suggestions!

Either `_` or `_MVVMToolkit_` is fine. But if you think they don't need to be unique then select the former.

Nirmal4G commented:
There's no official docs on this as far as I know. It's just a convention (_doesn't actually make your property internal/private_) to make sure we don't accidentally use potential MSBuild properties that could be used by users.

You can refer to other props and targets shipped by M$FT and you can see this pattern even in Common props and targets.

Nirmal4G commented:
There's no official docs on this as far as I know. It's just a convention (_doesn't actually make your property internal/private_) to make sure we don't accidentally use potential MSBuild properties that could be used by users.

You can refer to other props and targets shipped by M$FT and you can see this pattern even in Common props and targets.

Sergio0694 commented:
I see. In that case I don't think we need that, just using "MVVMToolkit" as prefix should be enough. This has been the original suggestion from Sam as well (from the Roslyn team), and what he's been doing as well in the [Antlr4cs project](https://github.com/tunnelvisionlabs/antlr4cs/blob/master/runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets) too.

Nirmal4G commented:
Okay, he doesn't seem to use the prefix even for properties that shouldn't be available to user. I'm okay with that. It looks neat. But you do have targets name prefixed. And it's inconsistent with elsewhere, not that I'm complaining but yeah...😅

Sergio0694 commented:
That's a fair point, I think Sam must've also changed his mind on this as he was the one that used "_" in the first place.
I've updated the file in https://github.com/CommunityToolkit/dotnet/pull/321/commits/86aaa9953dbd20ad33a47bba8f76e3cb34af0346 and removed all "_" prefixes, now we just use "MVVMToolkit___" everywhere 🙂

Nirmal4G reviewed (approved):
Nice... Looks good!
